--- XboxTokenManager.original.js	2024-04-13 22:48:48.715468953 +0200
+++ XboxTokenManager.js	2024-04-13 22:42:45.305531004 +0200
@@ -3,7 +3,7 @@
 const XboxLiveAuth = require('@xboxreplay/xboxlive-auth')
 const debug = require('debug')('prismarine-auth')
 const { SmartBuffer } = require('smart-buffer')
-const { exportJWK } = require('jose')
+const { exportJWK, importSPKI } = require('jose')
 const fetch = require('node-fetch')
 
 const { Endpoints, xboxLiveErrors } = require('../common/Constants')
@@ -22,8 +22,10 @@
 class XboxTokenManager {
   constructor (ecKey, cache) {
     this.key = ecKey
-    exportJWK(ecKey.publicKey).then(jwk => {
-      this.jwk = { ...jwk, alg: 'ES256', use: 'sig' }
+    importSPKI(ecKey.publicKey.export({ type: 'spki', format: 'pem' }), 'ES256', { extractable: true }).then(pubKey => {
+      exportJWK(pubKey).then(jwk => {
+        this.jwk = { ...jwk, alg: 'ES256', use: 'sig' }
+      })
     })
     this.cache = cache
 
